test/e2e/helm_out_of_band.go

```diff
@@ -0,0 +1,104 @@
+	WaitPeriod            time.Duration                  // The time to wait before verifying that the Helm release is unchanged.
+}
+
+// HelmReleaseOutOfBandSpec implements a test that verifies an existing Helm release will not be changed on the InstallOnce strategy when the HelmChartProxy
+// changes or the cluster is unselected. It assumes that the Helm release was installed successfully prior and verifies that the Helm release is unchanged.
+func HelmReleaseOutOfBandSpec(ctx context.Context, inputGetter func() HelmReleaseOutOfBandInput) {
+	var (
+		specName   = "helm-upgrade"
+		input      HelmReleaseOutOfBandInput
+		mgmtClient ctrlclient.Client
+		err        error
+	)
+
+	input = inputGetter()
+	Expect(input.BootstrapClusterProxy).NotTo(BeNil(), "Invalid argument. input.BootstrapClusterProxy can't be nil when calling %s spec", specName)
+	Expect(input.Namespace).NotTo(BeNil(), "Invalid argument. input.Namespace can't be nil when calling %s spec", specName)
+
+	By("creating a Kubernetes client to the management cluster")
+	mgmtClient = input.BootstrapClusterProxy.GetClient()
+	Expect(mgmtClient).NotTo(BeNil())
+
+	// Get existing HCP from management Cluster
+	existing := &addonsv1alpha1.HelmChartProxy{}
+	key := types.NamespacedName{
+		Namespace: input.HelmChartProxy.Namespace,
+		Name:      input.HelmChartProxy.Name,
+	}
+	err = mgmtClient.Get(ctx, key, existing)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get the inital Helm release through the HelmChartProxy
+
+	// Wait for initial HelmReleaseProxy to be ready
+	hrpWaitInput := GetWaitForHelmReleaseProxyReadyInput(ctx, bootstrapClusterProxy, input.ClusterName, *input.HelmChartProxy, 1, specName)
+	WaitForHelmReleaseProxyReady(ctx, hrpWaitInput, e2eConfig.GetIntervals(specName, "wait-helmreleaseproxy-ready")...)
+
+	// Get workload Cluster proxy
+	By("creating a clusterctl proxy to the workload cluster")
+	workloadClusterProxy := bootstrapClusterProxy.GetWorkloadCluster(ctx, input.Namespace.Name, input.ClusterName)
+	Expect(workloadClusterProxy).NotTo(BeNil())
+
+	// Wait for Helm release on workload cluster to have status = deployed
+	existingReleaseWaitInput := GetWaitForHelmReleaseDeployedInput(ctx, workloadClusterProxy, hrpWaitInput.HelmReleaseProxy.Spec.ReleaseName, hrpWaitInput.HelmReleaseProxy.Spec.ReleaseNamespace, specName)
+	existingRelease := WaitForHelmReleaseDeployed(ctx, existingReleaseWaitInput, e2eConfig.GetIntervals(specName, "wait-helm-existingRelease-deployed")...)
+
+	// Invoke UpgradeHelmChart logic here
+	options := &HelmOptions{
+		Values: input.Values,
+	}
+	UpgradeHelmChart(ctx, workloadClusterProxy, existingRelease.Namespace, input.HelmChartProxy.Spec.RepoURL, existingRelease.Chart.Name(), existingRelease.Name, options, existingRelease.Chart.Metadata.Version)
+	// TODO: how do we figure out that the Helm release won't change? Trigger reconcililation maybe?
```

Comment:
Is this `// TODO` something we should figure out before merge, or is it ok to go with just a sleep period?

Reply:
I removed it as a TODO, but I think it's worth following up on if we later come up with a better idea on how to prove the Helm release won't change.